agent/src/util/config_action.rs

```diff
@@ -276,6 +291,24 @@ async fn handle_config_delete(
     Ok(())
 }
 
+/// Checks to see if the configuration needs to be recreated.
+/// At present, this just checks to see if the `.metadata.generation` has changed.
+/// The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+async fn should_recreate_config(config: &Configuration, config_map: ConfigMap) -> Result<bool, ()> {
```

Comment:
I chose to put this all in its own method so that in the future, if we decided to do something more precise than just looking at the generation, we do that more easily here so the event handler doesn't become too bloated.

Reply:
Awesome!